core/src/rpc_subscriptions.rs

```diff
@@ -144,21 +145,32 @@ where
     let mut notified_set: HashSet<SubscriptionId> = HashSet::new();
     if let Some(hashmap) = subscriptions.get(hashmap_key) {
         for (sub_id, (sink, confirmations)) in hashmap.iter() {
-            let desired_slot: Vec<u64> = current_ancestors
-                .iter()
-                .filter(|(_, &v)| v == *confirmations)
-                .map(|(k, _)| k)
-                .cloned()
-                .collect();
-            let root: Vec<u64> = current_ancestors
+            let root = current_ancestors
```

Comment:
nit: (pre-dates your changes) but this doesn't need to be recomputed for every subscription